catalogue/config.py

```diff
@@ -0,0 +1,122 @@
+import os.path
+import csv
+import argparse
+from argparse import Namespace
+import yaml
+from .utils import read_config_file, CONFIG_LOC, dictionary_printer
+
+def config_validator(config_loc):
+
+    """
+    Validate the format of a configuration file.
+
+    Applies the following checks, all of which are needed for the file to be
+    validated:
+
+    - config file can be read as a dictionary
+    - argument keys are a subset of (`--input_data`, `--code`, `--output_data`, `-csv`
+    `catalogue_results`)
+    - csv options specify a csv file
+    - argument values are all strings
+
+    config_loc specifies the location of the config file. Commands that involve
+    existing config files (such as the parser), will only use a validated config file.
+
+
+    Parameters:
+    ------------
+    config_loc : str
+
+    Returns:
+    ---------
+    Boolean indicating if a given config file is validated
+
+    """
+
+    config_dict = read_config_file(config_loc)
+
+    valid = True
+
+    # check config_dict is in fact a dictionary. The other checks require it to be a dictionary
+    if isinstance(config_dict, dict):
+
+        # checks that the config keys are a subset of the correct ones
+        valid_keys = ['catalogue_results','code','csv','input_data','output_data']
+        if not set(config_dict.keys()).issubset(valid_keys):
+            valid = False
+            print('Config error: invalid keys present in the yaml file')
+
+        # check that any csv configurations have the .csv extension
+        if 'csv' in config_dict.keys() and config_dict['csv'] is not None:
+            filename = config_dict['csv']
+
+            #change to use os path
+            extension = filename[-4:]
+            if extension != '.csv':
+                valid = False
+                print('Config error: csv argument has an invalid extension')
+
+        # check that all config file keys only have string values (i.e. no nested)
+        values_list = list(config_dict.values())
+        for value in values_list:
+            if not isinstance(value, str) and value is not None:
```

Comment:
I think (don't quote me on this as it is from memory) you can provide a tuple of types to `isinstance` and it will return True if it is one of them. Probably would clean this up a bit and make the logic more clear.